frame/system/src/lib.rs

```diff
@@ -493,6 +493,26 @@ pub mod pallet {
 			Self::deposit_event(Event::Remarked { sender: who, hash });
 			Ok(().into())
 		}
+
+		/// Make some on-chain remark and emit event.
+		///
+		/// # <weight>
+		/// - `O(b)` where b is the length of the remark.
+		/// - 1 event.
+		/// # </weight>
+		#[pallet::weight(T::SystemWeightInfo::remark_with_event(remark.len() as u32))]
+		//#[pallet::weight(T::SystemWeightInfo::remark_with_index(remark.len() as u32))]
+		pub fn remark_with_index(
```

Comment:
Do we really want to have this part of `frame-system`?

This will bleed into any chain using Substrate, including Polkadot/Kusama.

Reply:
This is not a standalone chain, but a substrate feature indeed. @gavofyork Suggested it should be enabled on kusama and eventually polkadot.

Reply:
Yeah, but that could also be done by a special pallet. 

I mean parachains will then also get this functionality etc. IMO we should not enable this by default everywhere. Yes, everybody could filter the call, but that sounds more like a hack to me.

Reply:
Makes sense. I'll move it to a separate pallet.